module/zfs/btree.c

```diff
@@ -205,6 +214,7 @@ zfs_btree_create_custom(zfs_btree_t *tree,
 {
 	size_t esize = lsize - offsetof(zfs_btree_leaf_t, btl_elems);
 
+	VERIFY3U(size, <=, BTREE_ELEM_MAX_SIZE);
```

Comment:
nit: This seems like an ASSERT3U should be fine

Reply:
Leaving this as VERIFY seems reasonable to me just to make absolutely sure we catch an incorrect use even with production builds.